controllers/authorino_controller.go

```diff
@@ -126,11 +132,29 @@ func (r *AuthorinoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		// Deployment created successfully - return and requeue
 		return ctrl.Result{Requeue: true}, nil
 	} else {
+
+		if !deploymentAvailable(existingDeployment) {
+			// Deployment not ready – return and requeue
+			err = updateStatusConditions(logger, authorinoInstance, r.Client, statusNotReady(statusDeploymentNotReady, "Authorino Deployment resource not ready"))
+			return ctrl.Result{RequeueAfter: time.Second}, err
+		}
+
 		// deployment already exists, then build a new resource with the desired changes
 		// and compare them, if changes are encountered apply the desired changes
-		desiredDeployment := r.buildAuthorinoDeployment(authorinoInstance)
-		logger.Info("desiredDeployment", "deployment", desiredDeployment)
-		logger.Info("existingDeployment", "deployment", existingDeployment)
+		desiredDeployment, err := r.buildAuthorinoDeployment(authorinoInstance)
+		// merge existing key/value pairs back into the desiredDeployment Spec Template Labels so as not to overwrite those.
+		for existingKey, existingValue := range existingDeployment.Spec.Template.Labels {
+			if desiredValue, exists := (desiredDeployment.Spec.Template.Labels)[existingKey]; !exists || desiredValue != existingValue {
```

Comment:
what if the user adds a label which key conflicts with desired one? for instance `authorino-resource: something else` ?

Reply:
Can this happen on update? the deployment would already be created with the existing hard coded ones. so the only edit they could make is to the value. if they do that we want to set it back to the desired value right?

Reply:
Right, it is not about _adding_ but updating the value. If that happens, the pod templates would not match the selector and API server would reject the call so you would get an error on the update call, so you would try and try endlessly. Try it, as I might be wrong. If I am not wrong, I would add the logic to protect some label keys against changes existing in the PR of the limitador operator

Reply:
The selector labels are spec.selector.LabelSelector.MatchLabels. these are not edited in this case. so any hardcoded desired key will be put back to the value we expect. only if the desiredKey doesn't exist will an update be made. otherwise the value stays the same as what's in desired, which is the hardcoded ones.

Reply:
Sorry. Late for the party at https://github.com/Kuadrant/authorino-operator/pull/243#discussion_r1971854555, but sounds like @eguzki has a point. We must ensure pod labels match the deployment selector.